its/SonarCloudCallWrapper.cs

```diff
@@ -0,0 +1,135 @@
+﻿/*
+ * Azure DevOps extension for SonarQube/SonarCloud ITs
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using AzureDevOpsExtension.IntegrationTests.Models;
+using Newtonsoft.Json;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AzureDevOpsExtension.IntegrationTests
+{
+    public class SonarCloudCallWrapper
+    {
+        private string _sonarCloudToken => Environment.GetEnvironmentVariable("SC_TOKEN");
+        private string _sonarCloudBaseUrl => Environment.GetEnvironmentVariable("SC_BASE_URL");
+
+        private string GetBase64EncodedToken()
+        {
+            return Convert.ToBase64String(ASCIIEncoding.ASCII.GetBytes($"{_sonarCloudToken}:"));
+        }
+
+        private HttpClient GetHttpClient()
+        {
+            var httpClient = new HttpClient()
+            {
+                BaseAddress = new Uri(_sonarCloudBaseUrl),
+            };
+
+            httpClient.DefaultRequestHeaders.Add("Authorization", $"Basic {GetBase64EncodedToken()}");
+
+            return httpClient;
+        }
+
+        private async Task<MeasureBaseModel> GetNclocAndCoverageForComponent(string component)
+        {
+            try
+            {
+                var httpClient = GetHttpClient();
+
+                var httpResult = await httpClient.GetAsync($"/api/measures/component?component={component}&metricKeys=coverage,ncloc");
+
+                var resultContent = JsonConvert.DeserializeObject<MeasureBaseModel>(await httpResult.Content.ReadAsStringAsync());
+
+                return resultContent;
+
+            }
+            catch (Exception ex)
```

Comment:
I'm not sure catching and squashing the server exceptions is helpful; the test is still going to fail later (e.g. an assertion fails), but it will take more digging to work out why.
Test execution failures will be easier to diagnose if we let the test framework catch and log the underlying exception.

Reply:
Agreed, i'll remove those catches